MathlibTest/abel.lean

```diff
@@ -141,3 +141,22 @@ example [AddCommGroup α] (x y z : α) (h : False) (w : x - x = y + z) : False :
   abel_nf at *
   guard_hyp w : 0 = y + z
   assumption
+
+/-
+Test that when `abel_nf` normalizes multiple expressions which contain a particular atom, it uses a
+form for that atom which is consistent between expression.
+
+Note: This test is useless unless done at `=ₛ` (the level of syntactic equality); doing this test at
+`=` (the level of reducible equality) would not catch the erroneous old behaviour (normalizing one
+expression to `2 • a` and the other to `2 • x`).  But to get syntactic equality, we need to make the
```

Comment:
Is this not what instance reducibility is for?

Reply:
Sorry, I was confused about what was going on here. Indeed, instance reducibility doesn't help.